### PR TITLE
Add package_inventory note to packagesmatching and packageupdatesmatc…

### PR DESCRIPTION
**Pull request by pasinskim**
…hing.

Add description about package_inventory body common control attribute to
packagesmatching() and packageupdatesmatching() documentation.

### CODE REVIEW

**Comment by pasinskim** [comment]
@nickanderson can you check if this description makes sense?

**Comment by nickanderson** [comment]
This indicates that you can only report for a single package manager. For example I would have to choose between pip installed packages vs RPM installed packages. Is that the case or is this just for default package inventory?

**Comment by kacf** [comment]
`package_inventory` is a list, so you can query for whatever package managers are in the list.

**Comment by pasinskim** [comment]
Right. The thing is that you need to fill the ```package_inventory``` list with the package modules you want to be reported. If the list is empty then ```packagesmatching()``` won't work.

**Comment by kacf** [comment]
I suppose technically speaking it will work if you have package promises that use the same package module you're searching for, but this is not recommended, and I don't think we should even mention it. The official way is to use the `package_inventory`.

**Comment by pasinskim** [comment]
@kacf actually it won't. We are querying database to get ONLY packages from modules matching entries in ```package_inventory```.

**Comment by kacf** [comment]
Ok, I see. I thought these databases were shared.

**Comment by kacf** [comment]
The notes seem good to me, btw.

File: reference/functions/packagesmatching.markdown [hunk 1]
```diff
@@ -39,6 +39,8 @@ The return is a data container with a list of package descriptions, looking like
 The following code extracts just the package names, then looks for
 some desired packages, and finally reports if they are installed.
 
+*IMPORTANT:* Please note that you need to provide ```package_inventory``` attribute in ```body common control``` in order to be able to use this function. Also depending on the value(s) of ```package_inventory``` only packages from selected package modules will be returned. For more information about ```package_inventory``` please read [`package_inventory`][Components and Common Control#package_inventory] section.
```
Comment (nickanderson) [inline reference/functions/packagesmatching.markdown]: Double asterisks for bolding in markdown.
**IMPORTANT:**

Single backticks for verbatium or automatic linking to cfengine attributes:

`package_inventory` `body common control`

**Comment by pasinskim** [comment]
@nickanderson thanks for review. I've added ```refresh rules``` section describing when cache is updated for both functions.

File: reference/functions/packageupdatesmatching.markdown [hunk 2]
```diff
@@ -34,12 +34,23 @@ The return is a data container with a list of package descriptions, looking like
 * `arch_regex` - Regular expression matching package architecutre
 * `method_regex` - Regular expression matching package method (apt-get, rpm, etc ...)
 
+**IMPORTANT:** Please note that you need to provide ```package_inventory``` attribute in ```body common control``` in order to be able to use this function. Also depending on the value(s) of ```package_inventory``` only packages from selected package modules will be returned. For more information about ```package_inventory``` please read [`package_inventory`][Components and Common Control#package_inventory] section.
```
Comment (nickanderson) [inline reference/functions/packageupdatesmatching.markdown]: single *`* througout

**Comment by nickanderson** [comment]
looks pretty good, just replace the ```word``` with `word` except in the case of ```cf3 :)

File: reference/functions/packageupdatesmatching.markdown [hunk 3]
```diff
@@ -34,12 +34,23 @@ The return is a data container with a list of package descriptions, looking like
...
 ```cf3
       "all_package_updates" data => packageupdatesmatching(".*", ".*", ".*", ".*");
 ```
 
+**Refresh rules:**
+* updates cache used by packageupdatesmatching() is refreshed at the end of each agent run in accordance with constraints defined in the relevant package module body.
+* updates cache is refreshed every time ```repo``` type package is installed or removed
+* updates cache is refreshed if no local cache exists.
+        This means a reliable way to force a refresh of CFEngine's internal package cache is to simply delete the local cache:
+            $(sys.statedir)/packages_updates_<package_module>.lmdb*
```
Comment (kacf) [inline reference/functions/packageupdatesmatching.markdown]: Quote file name to give it monospace font. Use triple backtick (`) notation without the cf3.

File: reference/functions/packageupdatesmatching.markdown [hunk 4]
```diff
@@ -34,12 +34,23 @@ The return is a data container with a list of package descriptions, looking like
...
+**IMPORTANT:** Please note that you need to provide ```package_inventory``` attribute in ```body common control``` in order to be able to use this function. Also depending on the value(s) of ```package_inventory``` only packages from selected package modules will be returned. For more information about ```package_inventory``` please read [`package_inventory`][Components and Common Control#package_inventory] section.
+
+
 **Example:**
 
 ```cf3
       "all_package_updates" data => packageupdatesmatching(".*", ".*", ".*", ".*");
 ```
 
+**Refresh rules:**
```
Comment (kacf) [inline reference/functions/packageupdatesmatching.markdown]: Maybe also comment that network is only hit if the ifelapsed has expired, regardless of the rules below.

File: reference/functions/packagesmatching.markdown [hunk 5]
```diff
@@ -47,6 +49,13 @@ some desired packages, and finally reports if they are installed.
       "all_packages" data => packagesmatching(".*", ".*", ".*", ".*");
 ```
 
+**Refresh rules:** 
+* inastalled packages cache used by packagesmatching() is refreshed at the end of each agent run in accordance with constraints defined in the relevant package module body.
+* installed packages cache is refreshed after installing or removing a package.
+* installed packages cache is refreshed if no local cache exists.
+        This means a reliable way to force a refresh of CFEngine's internal package cache is to simply delete the local cache:
+            $(sys.statedir)/packages_installed_<package_module>.lmdb*
```
Comment (kacf) [inline reference/functions/packagesmatching.markdown]: Quote here as well, like the other one.

File: reference/functions/packageupdatesmatching.markdown (hunk 3 above)
Comment (nickanderson) [inline reference/functions/packageupdatesmatching.markdown]: Why triple backtick? I use single backticks usually.

Comment (nickanderson) [inline reference/functions/packageupdatesmatching.markdown]: Oh you mean

backtickbacktickbacktickcf3
vs 
backtickbacktickbacktick

for verbatim blocks. yes. tick tick tick cf3 is for cfengine code. verbatim single words can be single backtick though.

Comment (kacf) [inline reference/functions/packageupdatesmatching.markdown]: Yes, I meant as a block, not as a word.

**Pull request merged**